### PR TITLE
[Merged by Bors] - feat: add `LinearIsometryEquiv.piLpCongrRight`

### PR DESCRIPTION
**Pull request by eric-wieser**
Also relax some typeclasses on `piLpCongrLeft`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/PiLp.lean [hunk 1]
```diff
@@ -741,6 +742,58 @@ theorem _root_.LinearIsometryEquiv.piLpCongrLeft_single [DecidableEq ι] [Decida
...
+      Equiv.apply_symm_apply, WithLp.linearEquiv_symm_apply, WithLp.linearEquiv_apply]
+    obtain rfl | hp := p.dichotomy
+    · simp_rw [PiLp.norm_equiv_symm, Pi.norm_def, LinearEquiv.piCongrRight_apply,
+        LinearIsometryEquiv.coe_toLinearEquiv, LinearIsometryEquiv.nnnorm_map]
+    · have : 0 < p.toReal := zero_lt_one.trans_le <| by norm_cast
+      simp only [PiLp.norm_eq_sum this, WithLp.equiv_symm_pi_apply, LinearEquiv.piCongrRight_apply,
+        LinearIsometryEquiv.coe_toLinearEquiv, LinearIsometryEquiv.norm_map]
+
+@[simp]
+theorem _root_.LinearIsometryEquiv.piLpCongrRight_apply (e : ∀ i, α i ≃ₗᵢ[𝕜] β i) (x : PiLp p α):
```
Comment (riccardobrasca) [inline Mathlib/Analysis/NormedSpace/PiLp.lean:772]:
```suggestion
theorem _root_.LinearIsometryEquiv.piLpCongrRight_apply (e : ∀ i, α i ≃ₗᵢ[𝕜] β i) (x : PiLp p α) :
```
Can `simps` generate this?

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Analysis/NormedSpace/PiLp.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/PiLp.lean]: Not without some refactoring of existing simp lemmas it can't.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**